client/js/table-model.js

```diff
@@ -17,6 +23,121 @@ class TableModel {
 		this.data[this._getCellId(location)] = value;
 	}
 
+	getColor(location) {
+		return this.colors[this._getCellId(location)];
+	}
+
+	setColor(location, color) {
```

Comment:
cell color should absolutely not be in the Model of the spreadsheet.  The  model is for the data, not how it's displayed.

Cell coloring a rendering concern of the View, and should be done with CSS, not in JS code.